### ISSUE TITLE
Orca does not announce the file name when using ctrl + PageUp or ctrl + PageDown to switch between files

### ISSUE THREAD

**Issue by jvesouza** [microsoft/vscode#95529]
Issue Type: <b>Bug</b>

1. Launch orca.
2. Open at least two files in VSCode.
3. Switch between files using ctrl + Page Up or ctrl + page Down.

Instead of announcing the name of the file orca seems to read the contents of the file present in the new window.
CC @isidorn  
VS Code version: Code - Insiders 1.45.0-insider (a3083e58e2a9e77f60e531acaa32e18527ae8439, 2020-04-17T08:09:45.292Z)
OS version: Linux x64 5.6.4-arch1-1

**Comment by isidorn** [comment]
@jvesouza I tried reproducing this and I could not. Let me know if you are doing something special.
Orca nicely reads the title of the editor when I navigate to it.

@joanmarie @webczat can you give this and try to repro?

**Comment by jvesouza** [comment]
@isidorn  The file name is actually announced by orca, but only after all the content of the file is read.
In my environment, when I press ctrl + pageUp or ctrl + pageDown, the orca reads all the contents of the file and then announces its name.

**Comment by isidorn** [comment]
Orca should never read the whole file content, only the current line, and only after the title of the editor is read. That's weird.

**Comment by joanmarie** [comment]
Confirmed. What happens each time you switch page tabs is Orca gets a text insertion event for the entire contents of the text. I wish it didn't get that event, but....

When Orca gets text-insertion events, a series of heuristics kick in where Orca plays "guess why the insertion happened" in an attempt to determine if the text got inserted for a reason that suggests the user should be told about it (e.g. due to performing a command, e.g. in a terminal). Orca is obviously guessing wrong here. ;) I'll fix this bug in Orca.

**Comment by joanmarie** [comment]
@jvesouza: Please pull Orca master and re-test. Thanks!

**Comment by jvesouza** [comment]
@joanmarie  I tested and confirm that it has been fixed.
One thing I noticed while testing is that at least on my machine there is a delay of about 1 second for the orca to announce the new file.
This small delay happens when I switch using ctrl + pageup / down or alt + N.

**Comment by joanmarie** [comment]
> @joanmarie I tested and confirm that it has been fixed.

Yay!

> This small delay happens when I switch using ctrl + pageup / down or alt + N.

Mind emailing me a debug.out? There might be some other event spam that I need to filter out. Or ... I won't know until I see your output.

**Comment by isidorn** [comment]
Ok. This seems fixed. Thanks a lot @joanmarie thus closing this.

**Comment by jvesouza** [comment]
Mind emailing me a debug.out? There might be some other event spam that 
I need to filter out. Or ... I won't know until I see your output.
>
> —
>
I sent the debug file in a private message. Please let me know if you 
haven't received it. Thank you.

**Comment by joanmarie** [comment]
@jvesouza I got it. Thanks!

**Comment by joanmarie** [comment]
@jvesouza Sorry for the delay, but I think I managed to shave off some processing time and, at least in my environment, some chattiness. Could you please pull Orca master and re-test? If there's still a delay, please send me a new debug.out. Thanks!
